Remotion/ObjectBinding/Core/Validation/IBusinessObjectValidationResult.cs

```diff
@@ -23,6 +23,11 @@ namespace Remotion.ObjectBinding.Validation
   /// <threadsafety static="true" instance="false" />
   public interface IBusinessObjectValidationResult
   {
+    IReadOnlyCollection<BusinessObjectValidationFailure> GetValidationFailuresForBusinessObject (
+        [NotNull] IBusinessObject businessObject,
+        bool includePartiallyHandledFailures = false,
+        bool markAsHandled = true);
+
```

Comment:
The new API doesn't fit into the existing contract:

Naming: I think `GetValidationFailures` should be sufficient considering the existing methods on this interface. Not specifying a BusinessObjectProperty implies we get all failures for the BusienssObject, regardless of the property (see `GetUnhandledValidationFailures`)

`GetValidationFailuresForBusinessObject` should not have a parameter `includePartiallyHandledFailures`. Instead, it will always return all validation failures. If, on the other hand, the intention is to get unhandled failures, we should add this into the name, see `GetUnhandledValidationFailures` and drop `markAsHandled`. 

It's very likely that this comment doesn't cover the actually intended scenario / design and we need to discuss how to best design the API so it works within the parameters of the existing API or how the existing API should/need to be updated.

Reply:
I discussed this issue with mike and we create a new design for the functionality we need. Right now, the method is very orthogonal to the rest of the design. To fix this, we are going to extend the `GetUnhandledValidationFailures` methods with the `markAsHandled` functionality.

Another problem is the fact that `GetUnhandledValidationFailures` returns an `UnhandledBusinessObjectValidationFailure` and not a `BusinessObjectValidationFailure` that we need. We decided to simply change the return type here as the same properties are available so the only break are explicit variable type usages.

Our current implementation of `GetValidationFailuresForBusinessObject` should already do what we want, so we can use it to replace the `GetUnhandledValidationFailures` method. Please check again if the logic matches up and we don't break anything by replacing the method. We'll also need a breaking change ticket to document the changed return type of the `GetUnhandledValdiationFailures` method.